src/CreeDictionary/CreeDictionary/paradigm/manager.py

```diff
@@ -103,6 +114,37 @@ def _load_layouts_from(self, path: Path):
             # exist yet:
             self._name_to_layout.setdefault(paradigm_name, {})[size] = layout
 
+    _LITERAL_LEMMA = re.compile(r"\$\{lemma\}")
+
+    @cache
+    def all_analysis_template_tags(self, paradigm_name) -> Collection[tuple]:
+        """Return the set of all analysis templates in layouts of paradigm_name
+
+        If a paradigm has two sizes, one with template `${lemma}+A` and the
+        other with both `${lemma}+A` and `X+${lemma}+B`, then this function will
+        return {((), ("+A",)), (("X+",), ("+B",)}.
+
+        Note that these analyses are meant to be inputs to a generator FST for
+        building a paradigm table, not the results of analyzing some input
+        string.
+        """
+        ret = {}
+        for layout in self._name_to_layout[paradigm_name].values():
+            # The trick here is that we can look for a literal `${lemma}`
+            # instead of having to parse arbitrary FST analyses.
+            for template in layout.generate_fst_analyses("${lemma}"):
+                prefix, suffix = self._LITERAL_LEMMA.split(template)
+
+                prefix_tags = prefix.split("+")
+                assert prefix_tags[-1] == "", f"Prefix ${prefix!r} did not end with +"
+                suffix_tags = suffix.split("+")
+                assert suffix_tags[0] == "", f"Suffix ${suffix!r} did not end with +"
+                ret[template] = (
+                    tuple(t + "+" for t in prefix_tags[:-1]),
+                    tuple("+" + t for t in suffix_tags[1:]),
+                )
```

Comment:
It will work fine for Tsuut’ina. For Arapaho, this code will need to be adjusted since that FST uses `[TAG]` instead of `+Tag`.

Reply:
Is this reference to the + signs because (somewhere) within itwêwina tags were stilled represented as a single squashed string, rather than as distinct multichar symbols when that is the case?

Reply:
There are two things in the code that jump to mind right now and still use smushed strings:

  - The paradigm files themselves have smushed strings like `${lemma}+V+AI+Ind+12Pl`. The code here parses that into distinct multichar symbols. The generator FST doesn’t need them separated, but the python code does to show the right tags. A more general but slower method to get this info would be to (1) pass smushed string to strict generator (2) pass generated wordform to strict analyzer, getting multichar symbols back (3) when multiple possibilities are returned by step 2, choose the one whose smushing matches the original from step #1.

  - `crk.altlabel.tsv` has tags with the `+` stripped, e.g., `1Sg` and `2Pl`, as well as smushed things like `0Sg+Fut+Def`. The code that deals with relabelling operates on these tags with the `+` signs removed, and there is code in this PR that has to strip off the `+` signs for compatibility with that.

    Switching the altlabel file to use full tags, maybe with a comma separator like `+0Sg,+Fut,+Def` to make it unambiguous which tag tag the `+` belongs to , is something that would be useful down the road.